Library/Homebrew/cask/lib/hbc/cask.rb

```diff
@@ -85,6 +85,45 @@ def installed?
       !versions.empty?
     end
 
+    def outdated?(greedy = false)
+      greedy = true if greedy
+      @outdated ||= {}
+      @outdated[greedy] ||= _check_outdated(greedy)
+      !@outdated[greedy].empty?
+    end
+
+    def outdated_versions(greedy = false)
+      greedy = true if greedy
+      @outdated ||= {}
+      @outdated[greedy] ||= _check_outdated(greedy)
+    end
+
+    def _check_outdated(greedy = false)
+      # special case: tap version is not available
+      return [] if version.nil?
+
+      # if --greedy is not present in the command line then skip when detection is not safe
+      unless greedy
+        return [] if auto_updates
+        return [] if version.latest?
+      end
+
+      # keep the installed versions in a local variable because a call to :versions is expensive
+      installed = versions
+      current   = installed.last
+
+      # if version is "latest" then all installed version are outdated (greedy)
+      return installed if version.latest?
+
+      # not outdated unless there is a different version on tap
+      return [] unless current != version
+
+      # current installed version different than tap version
+      # collect all installed versions that are different than tap version and return them
+      installed.map { |v| v if v != version.to_s }.compact
```

Comment:
`#select` is the function I was looking for (but it seems I was too lazy to read the complete list of `Array` methods and I concluded too early). Thank you!